libcobblersignatures/__init__.py

```diff
@@ -2,3 +2,221 @@
+
+    def _importsignaturesurl(self, url: str):
+        """
+        Internal function to handle the import of a URL based import. The data behind it should be a well formed JSON.
+
+        :param url: The URL to load the content from. This parameter is handed to urllib.
+        """
+        response = urllib.request.urlopen(url)
+        data = response.read()
+        self.signaturesjson = data.decode('utf-8')
+
+    def __prepare_export_output(self, sort_keys: bool = False, indent: Union[None, int] = None) -> str:
+        """
+        Convert the internal data to a JSON. Only for internal usage.
+
+        :param sort_keys: If the keys of the dictionary should be sorted to be more human readable.
+        :param indent: If this is something other then ``None`` then the JSON will be pretty printed.
+        """
+        value = {}
+        for breed in self.osbreeds:
+            value[breed.name] = breed.encode()
+        return json.dumps({self._rootkey: value}, sort_keys=sort_keys, indent=indent)
+
+    def exportsignatures(self, export_type: ExportTypes, target: str = "", sort_keys: bool = False,
+                         indent: Union[None, int] = None):
+        """
+        This is the main export function.
+
+        :param export_type: One of the values from the :class:`ExportTypes`.
+        :param target: This is only required when using this for a file based export. Otherwise this can be skipped.
+        :param sort_keys: If the keys of the dictionary should be sorted to be more human readable.
+        :param indent: If this is something other then ``None`` then the JSON will be pretty printed.
+        :raises ValueError: When the :class:`ExportTypes` is not implemented or not known.
+        :raises TypeError: When one of the arguments has the wrong type.
+        """
+        if not isinstance(target, str):
+            raise TypeError("target needs to be of type str!")
+        if not isinstance(sort_keys, bool):
+            raise TypeError("sort_keys needs to be of type bool!")
+        if not (indent is None or isinstance(indent, int)):
+            raise TypeError("indent needs to be of type integer or None!")
+
+        if export_type == ExportTypes.FILE:
+            if not target:
+                raise ValueError("Please provide a path if your want to export to a file!")
+            with open(target, "w") as f:
+                f.write(self.__prepare_export_output(sort_keys, indent))
+        elif export_type == ExportTypes.STRING:
+            return self.__prepare_export_output(sort_keys, indent)
+        else:
+            raise ValueError("Please use on of the two given options for the export type!")
+
+    def jsontomodels(self):
+        """
+        Convert the loaded JSON to the internal modules. Without calling this the loaded data will not be available for
+        manipulation.
+        """
+        breeds = self.signaturesjson.get(self._rootkey, -1)
+        if breeds == -1:
```

Comment:
[nitpick] `-1` could be a self-documenting thing, e.g.
```python
>>> Missing = object()
>>> breeds = {}.get("foo", Missing)
>>> if breeds is Missing:
...     print("No breeds!")
... else:
...     print(f"Here are your breeds: {breeds}")
... 
No breeds!
```

You can read more about [this pattern here](https://python-patterns.guide/python/sentinel-object/).